app/Domain/Coupons/Repositories/CouponRepository.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Coupons\Repositories;
+
+use App\Domain\Coupons\Coupon;
+use Ramsey\Uuid\UuidInterface;
+
+interface CouponRepository
+{
+    public function save(Coupon $coupon): void;
+
+    public function getById(UuidInterface $id): Coupon;
+
+    public function getByIntegrationId(UuidInterface $integrationId): ?Coupon;
```

Comment:
This can be typed on `Coupon`, because the implementation will throw a `ModelNotFoundException`